olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.h

```diff
@@ -75,6 +77,7 @@ class VersionedLayerClientImpl {
  private:
   client::HRN catalog_;
   std::string layer_id_;
+  boost::optional<int64_t> catalog_version_;
```

Comment:
Not sure if it is that easy.
We need to be able to set this from an async task and protect against concurrent tasks. Atomic maybe?

Reply:
I would postpone this task later, when we implement the actual logic behind `boost::none` and `GetLatestVersion`. Maybe implementor will use the `std::mutex` to protect data - this way we can omit `std::atomic` if we guarantee that all the access to `catalog_version_` would be read-only (correct me if I'm wrong).

Reply:
Then I would go the other way around.
First, implement logic in VersionedLayerClientImpl and then only connect.
Logic for now should be that if user specifies boost::none in request structures we take the catalog_version_ instead. If user specified version in request structures we assume he knows what he is doing and request with user version. After this logic is done we connect with public constructor and deprecate the WithVersion( ) from the requests in the same commit.